pymeasure/instruments/agilent/agilent33500.py

```diff
@@ -48,6 +46,317 @@
+    )
+
+    arb_filter = Instrument.control(
+        "SOUR{ch}:FUNC:ARB:FILT?",
+        "SOUR{ch}:FUNC:ARB:FILT %s",
+        """ A string property that selects the filter setting for arbitrary signals.
+        Can be set to 'NORM<AL>', 'STEP' and 'OFF'. """,
+        validator=strict_discrete_set,
+        values=["NORM", "NORMAL", "STEP", "OFF"],
+    )
+
+    arb_srate = Instrument.control(
+        "SOUR{ch}:FUNC:ARB:SRAT?",
+        "SOUR{ch}:FUNC:ARB:SRAT %f",
+        """ An floating point property that sets the sample rate of the currently selected
+        arbitrary signal. Valid values are 1 µSa/s to 250 MSa/s (maximum range, can be lower
+        depending on your device). This can be set. """,
+        validator=strict_range,
+        values=[1e-6, 250e6],
+    )
+
+    def data_volatile_clear(self):
+        """
+        Clear all arbitrary signals from the volatile memory. This should be done if the same name
+        is used continuously to load different arbitrary signals into the memory, since an error
+        will occur if a trace is loaded which already exists in the memory.
+        """
+        self.write("SOUR{ch}:DATA:VOL:CLE")
+
+    def data_arb(self, arb_name, data_points, data_format="DAC"):
+        """
+        Uploads an arbitrary trace into the volatile memory of the device. The data_points can be
+        given as comma separated 16 bit DAC values (ranging from -32767 to +32767), as comma
+        separated floating point values (ranging from -1.0 to +1.0) or as a binary data stream.
+        Check the manual for more information. The storage depends on the device type and ranges
+        from 8 Sa to 16 MSa (maximum).
+        TODO: *Binary is not yet implemented*
+
+        :param arb_name: The name of the trace in the volatile memory. This is used to access the
+                         trace.
+
+        :param data_points: Individual points of the trace. The format depends on the format
+                            parameter.
+
+                            format = 'DAC' (default): Accepts list of integer values ranging from
+                            -32767 to +32767. Minimum of 8 a maximum of 65536 points.
+
+                            format = 'float': Accepts list of floating point values ranging from
+                            -1.0 to +1.0. Minimum of 8 a maximum of 65536 points.
+
+                            format = 'binary': Accepts a binary stream of 8 bit data.
+        :param data_format: Defines the format of data_points. Can be 'DAC' (default), 'float' or
+                            'binary'. See documentation on parameter data_points above.
+        """
+        if data_format == "DAC":
+            separator = ", "
+            data_points_str = [str(item) for item in data_points]  # Turn list entries into strings
+            data_string = separator.join(data_points_str)  # Join strings with separator
+            print(f"SOUR{{ch}}:DATA:ARB:DAC {arb_name}, {data_string}")
```

Comment:
I guess this leftover from debug code. Right?

Reply:
It seems to be from previous commits, but you are right, when tested with pytest -s it threw tons of data prints. Removed in https://github.com/pymeasure/pymeasure/pull/763/commits/3e71c1eee00e7ff48596b6418855cf6a67fe831e